<?xml version="1.0"?>
<launch>
  	<arg name="camera_name" default="camera" />
  	<!-- video_stream_provider can be a number as a video device or a url of a video stream -->
  	<arg name="video_stream_provider" default="0" />
  	
  	<!-- set buffer queue size of frame capturing to -->
  	<arg name="buffer_queue_size" default="100" />
  	<!-- frames per second to query the camera for -->
  	<arg name="fps" default="30" />
  	<!-- frame_id for the camera -->
  	
    <!-- force width and height -->
    <arg name="width" default="320"/>
    <arg name="height" default="240"/>

   	<!-- images will be published at /camera_name/image with the image transports plugins (e.g.: compressed) installed -->
   	
    <node pkg="amr-stream" type="publish.py" name="$(arg camera_name)_stream" output="screen"> 
        <param name="camera_name" type="string" value="$(arg camera_name)" />
        <param name="video_stream_provider" type="string" value="$(arg video_stream_provider)" />
        <param name="buffer_queue_size" type="int" value="$(arg buffer_queue_size)" />
        <param name="fps" type="double" value="$(arg fps)" />
        <param name="width" type="int" value="$(arg width)" />
        <param name="height" type="int" value="$(arg height)" />
    </node>

</launch>
